moonraker/components/file_manager/metadata.py

```diff
@@ -747,33 +747,65 @@ def parse_nozzle_diameter(self) -> Optional[float]:
 
 class IceSL(BaseSlicer):
     def check_identity(self, data) -> Optional[Dict[str, Any]]:
-        match = re.search(r"; <IceSL.*>", data)
+        match = re.search(r"<IceSL\s(.*)>", data)
         if match:
-            return {'slicer': "IceSL"}
+            version = match.group(1) if match.group(1)[0].isdigit() else "-"
+            return {
+                'slicer': "IceSL",
+                'slicer_version': version
+            }
         return None
 
     def parse_first_layer_height(self) -> Optional[float]:
         return _regex_find_first(
-            r"; z_layer_height_first_layer_mm :\s+(\d+\.\d+)",
+            r";\sz_layer_height_first_layer_mm\s:\s+(\d+\.\d+)",
             self.header_data)
 
     def parse_layer_height(self) -> Optional[float]:
         self.layer_height = _regex_find_first(
-            r"; z_layer_height_mm :\s+(\d+\.\d+)",
+            r";\sz_layer_height_mm\s:\s+(\d+\.\d+)",
             self.header_data)
         return self.layer_height
 
     def parse_object_height(self) -> Optional[float]:
-        return self._parse_max_float(
-            r"G0 F\d+ Z\d+\.\d+", self.footer_data, strict=True)
+        return _regex_find_first(
+            r";\sprint_height_mm\s:\s+(\d+\.\d+)", self.header_data)
 
     def parse_first_layer_extr_temp(self) -> Optional[float]:
         return _regex_find_first(
-            r"; extruder_temp_degree_c_0 :\s+(\d+\.?\d*)", self.header_data)
+            r";\sextruder_temp_degree_c_0\s:\s+(\d+\.?\d*)", self.header_data)
 
     def parse_first_layer_bed_temp(self) -> Optional[float]:
         return _regex_find_first(
-            r"; bed_temp_degree_c :\s+(\d+\.?\d*)", self.header_data)
+            r";\sbed_temp_degree_c\s:\s+(\d+\.?\d*)", self.header_data)
+
+    def parse_filament_total(self) -> Optional[float]:
+        return _regex_find_first(
+            r";\sfilament_used_mm\s:\s+(\d+\.\d+)", self.header_data)
```

Comment:
As explained in the 1st review point, this can be retrieved from the profile while extending the header.
For this particular information, see line 53 & 55 of the [Voron_V0](https://github.com/shapeforge/icesl-printers/blob/master/fff/Voron_V0/printer.lua) profile.